hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -307,6 +307,12 @@ object DefaultSource {
       val useNewParquetFileFormat = parameters.getOrElse(HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key(),
         HoodieReaderConfig.FILE_GROUP_READER_ENABLED.defaultValue().toString).toBoolean &&
         !metaClient.isMetadataTable && (globPaths == null || globPaths.isEmpty)
+      lazy val tableVersion = if (SparkConfigUtils.containsConfigProperty(parameters, INCREMENTAL_READ_TABLE_VERSION)) {
+        Integer.parseInt(parameters(INCREMENTAL_READ_TABLE_VERSION.key))
+      } else {
+        metaClient.getTableConfig.getTableVersion.versionCode()
+      }
+      lazy val hudiOneTable = tableVersion >= HoodieTableVersion.EIGHT.versionCode()
```

Comment:
can you add java docs to all out what exactly is `hudiOneTable`.lets not confused developers w/ One table or XTable :)

Reply:
Is 8 even the first 1.X table? Maybe there is a better name?